sct-commons/src/main/java/org/lfenergy/compas/sct/commons/ControlBlockEditorService.java

```diff
@@ -0,0 +1,384 @@
+                .flatMap(Collection::stream)
+                .filter(tRange -> tcbType.equals(tRange.getCBType()))
+                .findFirst()
+                .orElseThrow(() -> new ScdException("Control Block Communication setting files does not contain AppIdRange for cbType " + tcbType.value()));
+        PrimitiveIterator.OfLong appIdIterator = Utils.sequence(Long.parseLong(appIdRange.getStart(), 16), Long.parseLong(appIdRange.getEnd(), 16));
+
+        TRange macRange = Optional.ofNullable(cbCom.getMacRanges()).map(MacRanges::getMacRange).stream()
+                .flatMap(Collection::stream)
+                .filter(tRange -> tcbType.equals(tRange.getCBType()))
+                .findFirst()
+                .orElseThrow(() -> new ScdException("Control Block Communication setting files does not contain MacRange for cbType " + tcbType.value()));
+        Iterator<String> macAddressIterator = Utils.macAddressSequence(macRange.getStart(), macRange.getEnd());
+
+        Map<Criteria, Settings> settingsByCriteria = Optional.ofNullable(cbCom.getVlans()).map(Vlans::getVlan).stream()
+                .flatMap(Collection::stream)
+                .filter(vlan -> tcbType.equals(vlan.getCBType()))
+                .collect(Collectors.toMap(this::vlanToCriteria, this::vlanToSetting));
+
+        return new CbComSettings(appIdIterator, macAddressIterator, settingsByCriteria);
+    }
+
+    private Optional<SclReportItem> configureControlBlockNetwork(TCommunication tCommunication, Settings settings, PrimitiveIterator.OfLong appIdIterator, Iterator<String> macAddressIterator, TControl tControl, IedApLd iedApLd) {
+        if (settings.vlanId() == null) {
+            return newError(iedApLd, tControl, "Cannot configure communication for this ControlBlock because no Vlan Id was provided in the settings");
+        }
+        if (!appIdIterator.hasNext()) {
+            return newError(iedApLd, tControl, "Cannot configure communication for this ControlBlock because range of appId is exhausted");
+        }
+        if (!macAddressIterator.hasNext()) {
+            return newError(iedApLd, tControl, "Cannot configure communication for this ControlBlock because range of MAC Address is exhausted");
+        }
+
+        Optional<TConnectedAP> optConApAdapter = connectedApService.findConnectedAp(tCommunication, iedApLd.ied.getName(), iedApLd.apName);
+        if (optConApAdapter.isEmpty()) {
+            return newError(iedApLd, tControl, "Cannot configure communication for ControlBlock because no ConnectAP found for AccessPoint");
+        }
+        TConnectedAP tConnectedAP = optConApAdapter.get();
+        List<TP> listOfPs = new ArrayList<>();
+        listOfPs.add(newP(APPID_P_TYPE, Utils.toHex(appIdIterator.nextLong(), APPID_LENGTH)));
+        listOfPs.add(newP(MAC_ADDRESS_P_TYPE, macAddressIterator.next()));
+        listOfPs.add(newP(VLAN_ID_P_TYPE, Utils.toHex(settings.vlanId(), VLAN_ID_LENGTH)));
```

Comment:
`vlanId` is not null here because there is a check a the beginning of this method to ensure vlanId is not null. It is a requirement from #259. So I removed the case where vlanId is null.
```java
if (settings.vlanId() == null) {
    return newError(iedApLd, tControl, "Cannot configure communication for this ControlBlock because no Vlan Id was provided in the settings");
}
```
Extract of #259 : 
>  If for the current sextuplet, the settings file returns a VLAN-ID set to "None" then a fonctionnal error ERR.07 is returned

Reply:
ok